components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -124,11 +124,8 @@ type rtkSerial struct {
 	cancelFunc func()
 
 	activeBackgroundWorkers sync.WaitGroup
+	mu                      sync.Mutex
 
-	mu                 sync.Mutex // Mutex for general synchronization during reconfigure.
-	ntripMu            sync.Mutex // Mutex for NTRIP-related operations.
-	ntripconfigMu      sync.Mutex // Mutex for NTRIP configuration.
-	urlMutex           sync.Mutex // Mutex for URL-related operations.
```

Comment:
@penguinland did you test this? Removing urlMutex may cause some issues since this was used to prevent multiple function accessing ntrip buffer.

Reply:
yup, we tested it together! Things sure look like they're working correctly.